llvm/include/llvm/Frontend/Offloading/Utility.h

```diff
@@ -159,6 +160,107 @@ namespace intel {
 /// the Intel runtime offload plugin.
 Error containerizeOpenMPSPIRVImage(std::unique_ptr<MemoryBuffer> &Binary);
 } // namespace intel
+
+namespace sycl {
+class PropertySetRegistry;
+
+// A property value. It can be either a 32-bit unsigned integer or a byte array.
+class PropertyValue {
```

Comment:
I think that this class was overengineered from the very beginning.
This class can be replaced by simple `std::variant<uint64_t, std::string>`. In the text format it is not going to have much difference either it is uint64_t or uint32_t.